### PR TITLE
port to wahoo-style (new OMF) pkg

### PR DESCRIPTION
**Pull request by cap10morgan**
These changes were imported from my wahoo port of this plugin.

### CODE REVIEW

**Comment by cap10morgan** [comment]
Hmm, it doesn't seem to be adding the `functions` dir to `$fish_function_path` correctly. Maybe that `set` needs a new argument now that it's inside an init function?

**Comment by cap10morgan** [comment]
And/or `$fish_path` doesn't seem to exist (and it should be `pkg` instead of `plugins` now).

**Comment by cap10morgan** [comment]
Hmm, no, there's something more fundamental going wrong here. Looking into it.

**Comment by cap10morgan** [comment]
OK, seems to be working now.

File: emacs.fish [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    set __emacs_version (__major_version $__emacs)
+  end
+
+  if test "$__emacs_version" -gt 23
+    __set_editor
+    __add_functions_to_path
+  end
+
+  set -e emacs
+  set -e emacs_version
```
Comment (bobthecow) [inline emacs.fish]: are these trying to delete the `__emacs` and `__emacs_version` added above?

File: emacs.fish [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    set __emacs (which emacs)
+  end
+    
+  if not set -q __emacs_version
+    set __emacs_version (__major_version $__emacs)
+  end
+
+  if test "$__emacs_version" -gt 23
+    __set_editor
+    __add_functions_to_path
```
Comment (bpinto) [inline emacs.fish]: Too much indirection?
`set fish_function_path  $OMF_PATH/pkg/emacs/functions $fish_function_path` instead.

Comment (bpinto) [inline emacs.fish]: Nevermind.

Comment (cap10morgan) [inline emacs.fish]: That's how the original looked, and I took a "minimal changes to make it work approach."

File: emacs.fish (hunk 1 above)
Comment (cap10morgan) [inline emacs.fish]: This was also in the original, but it does seem odd. I assumed I just didn't understand what it was doing. If it should be `set -e __emacs` and `set -e __emacs_version` I'd be happy to change it.

Comment (bobthecow) [inline emacs.fish]: i imagine they were originally called `emacs` and `emacs_version` but were changed to make it more obvious that they were intended as local variables.

Comment (bpinto) [inline emacs.fish]: It's going to be delete anyway when the function finishes execution right? So we could remove these lines of code.

Comment (cap10morgan) [inline emacs.fish]: Yeah, they go out of scope at the end of the `init` function, right? I'll delete those lines.

File: emacs.fish (hunk 2 above)
Comment (bpinto) [inline emacs.fish]: @cap10morgan
Okay but it would be quite convenient to refactor this plugin as we can simplify it A LOT. :)

**Pull request merged**